tools/mpremote/mpremote/commands.py

```diff
@@ -334,6 +334,49 @@ def do_filesystem_recursive_rm(state, path, args):
             print(f"removed: '{path}'")
 
 
+def human_size(size, decimal_places=1):
+    for unit in ['B', 'K', 'M', 'G', 'T']:
+        if size < 1024.0 or unit == 'T':
+            break
+        size /= 1024.0
+    return f"{size:.{decimal_places}f} {unit}" if unit != 'B' else f"{int(size)} {unit}"
```

Comment:
Unix tree doesn't have a space between the number and unit, so maybe it doesn't need it here either?

Also, this won't work on boards without floating point support.  I think it's worth making it work on such boards.  Considering you only want 1 decimal place, you could multiply `size` by 10 to start with, then do `size // 10` and `size % 10` to format it.  Also, arguably you don't need the decimal if the integer part is >= 100 (only need 3 significant figures).

Reply:
> Also, this won't work on boards without floating point support.

This computation is not executed on the MCU though, or is it?

Reply:
> This computation is not executed on the MCU though, or is the intent for the host to lack FO capabilities?

Ah, you're right, this does not execute on the target MCU.

So, it's OK to use floats here.